cloudinit/net/networkd.py

```diff
@@ -296,12 +325,36 @@ def render_network_state(
 
         util.ensure_dir(network_dir)
 
-        ret_dict = self._render_content(network_state)
-        for k, v in ret_dict.items():
+        network = self._render_content(network_state)
+        vlan_netdev = network.pop("vlan_netdev", {})
+        bond_netdev = network.pop("bond_netdev", {})
+
+        for k, v in network.items():
             self.create_network_file(k, v, network_dir)
 
-    def _render_content(self, ns: NetworkState) -> dict:
+        for k, v in vlan_netdev.items():
+            self.create_network_file(k, v, network_dir, ext=".netdev")
+
+        for k, v in bond_netdev.items():
+            self.create_network_file(k, v, network_dir, ext=".netdev")
+
+    def _render_content(self, ns: NetworkState):
         ret_dict = {}
+        vlan_link = {}
+        bond_link = {}
+
+        if ns.version == 2:
```

Comment:
Does this mean only support vlans and bonds if using configuration v2? If so, why only v2?